src/pages/waiting-room-to-car/cat-home-test/components/vehicle-checks-modal/vehicle-checks-modal.cat-home-test.page.ts

```diff
@@ -73,39 +68,48 @@ export class VehicleChecksCatHomeTestModal {
     public store$: Store<StoreModel>,
     private navController: NavController,
     private faultCountProvider: FaultCountProvider,
-    questionProvider: QuestionProvider,
+    private testDataByCategoryProvider: TestDataByCategoryProvider,
+    private questionProvider: QuestionProvider,
   ) {
     this.formGroup = new FormGroup({});
-    this.showMeQuestions = questionProvider.getShowMeQuestions(TestCategory.F);
-    this.tellMeQuestions = questionProvider.getTellMeQuestions(TestCategory.F);
   }
 
   ngOnInit(): void {
     const currentTest$ = this.store$.pipe(
       select(getTests),
       select(getCurrentTest),
     );
+
+    currentTest$.pipe(
+      select(getTestCategory),
+    ).subscribe((value) => {
+      this.categoryCode = value;
+    }).unsubscribe();
```

Comment:
these don't feel right. I would have expected the unsubscribe to be in a view did leave